### PR TITLE
Process tracker api proposal

### PR DESCRIPTION
**Pull request by pasnox**
Add api to possibility track the inspected process state

Being is process running ? is process debugged ? is debugged process paused or running ?

### CODE REVIEW

File: common/processtracker_windows.cpp [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+
+    LocalBuffer(const LocalBuffer &other) = delete;
+    LocalBuffer &operator=(const LocalBuffer &other) = delete;
+
+    bool isValid() const {
+        return m_data != nullptr;
+    }
+
+    DWORD size() const {
+        return m_size;
```
Comment (giucam) [inline common/processtracker_windows.cpp]: If allocate() failed m_size will be different from 0 even if m_data is null, so maybe either set m_size to 0 there or check m_data here.

File: common/processtracker_macos.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        switch (info.kp_proc.p_stat) {
+            case SRUN:
+                pinfo.state = GammaRay::ProcessTracker::State::Running;
+                break;
+            case SSTOP:
+                pinfo.state = GammaRay::ProcessTracker::State::Suspended;
+                break;
+            default:
+                break;
+        }
```
Comment (giucam) [inline common/processtracker_macos.cpp]: Maybe print a warning if sysctl failed.

File: common/processtracker_macos.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    // Initialize mib, which tells sysctl the info we want, in this case
+    // we're looking for information about a specific process ID.
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PID;
+    mib[3] = pid;
+
+    // Call sysctl.
+    size = sizeof(info);
+    junk = sysctl(mib, sizeof(mib) / sizeof(*mib), &info, &size, NULL, 0);
```
Comment (pasnox) [inline common/processtracker_macos.cpp]: Done

File: common/processtracker_macos.cpp (hunk 2 above)
Comment (pasnox) [inline common/processtracker_macos.cpp]: Done

File: common/processtracker_windows.cpp (hunk 1 above)
Comment (pasnox) [inline common/processtracker_windows.cpp]: allocate() is private and can only result from resize() which first free() and set size to 0.
size it then updated only if ptr is not null.
Looks fine for me ?

Comment (giucam) [inline common/processtracker_windows.cpp]: Fair enough. Ok by me, then.

**Pull request merged**